### PR TITLE
[demo] bring in repokitteh's assign module

### PR DESCRIPTION
**Pull request by itayd**
*this PR is for demo purposes only and should not be merged. a replacement pr is now at https://github.com/envoyproxy/envoybot/pull/3.*

this is an example of how to bring in a ready-made repokitteh module. modules can also be coded inside this repo as well.

i'm going to invest heavily in docs very soon, this is just an example to show @cmluciano, @htuch and @mattklein123.

note that even while this change is not merged, it is still possible to test this in the pr itself. this might pose a security threat, or a way to subvert proper procedures so we might want to enable this to specific users, such as users which are members of a certain group or just a configurable whitelist.

also in the future i'm going to implement a security model where in the `use` clause one can specify what capabilities to allow the module. also the `use` clause can accomodate specifying a sha of the module to use.

### CODE REVIEW

**Comment by itayd** [comment]
from this point i merged a refactor into the original module (see https://github.com/softkitteh/repokitteh-modules/pull/2) so the behavior changes. since this `use` is not sha pinned (and it can be!), it will automatically take in the change.

**Comment by itayd** [comment]
\o/

**Comment by itayd** [comment]
/assign @scode  

i love nagging @scode.

**Comment by cmluciano** [comment]
This is really neat and the code looks impressively minimal. Is the sky code the only thing needed to intercept github hooks or does repokitteh have another backend service codebase?

**Comment by itayd** [comment]
The skylark code tells the backend what to do. The backend is the things the connects the skylark code with the github events processing. I should probably make a diagram or some slide deck explaining that. 

The whole idea is to allow a user to easily specify custom workflows/behaviors without the need to deal with all the backend. The frontend are the sky files - plus a UI when something fails (I can trigger a failure here to show).

Once I have an MVP (which one might argue I already have), I want to invest heavily in docs.

**Comment by itayd** [comment]
There ya go.

**Comment by htuch** [comment]
Neat! @itayd how do I test this?

**Comment by itayd** [comment]
@htuch what do you mean by testing? 
- unit testing? i will introduce this in the future (like `repokitteh_test.sky`).
- just trying it out? we can get this PR in and then I can show you how to write your own stuff. As I said, docs are coming soon...

So, what's next?

**Comment by htuch** [comment]
@itayd no, I mean try it out? How do I assign?

File: repokitteh.sky [hunk 1]
```diff
@@ -0,0 +1,5 @@
+use("github.com/softkitteh/repokitteh-modules/assign.sky")
```
Comment (htuch) [inline repokitteh.sky]: Yep, this seems pretty cool. I think there are a few things to things about:
1. We need better documentation, tutorial, HOWTOs etc. as you point out in your commit message.
2. Need to resolve trust issue here; still a bit nervous about granting write access to blackbox infrastructure. Might be better after understanding the full scope of what we're enabling with write access.
3. I'll chat with you on Slack about possible options for moving forward on the main repo, we should probably meet with CNCF folks such as @caniszczyk and figure out a good model.

Comment (itayd) [reply repokitteh.sky]: 1. Yup, of course. I want to solicit feedback from you good folks (as I'm doing now) to button stuff down, such as -
2. Yeah this is a big issue. There are multiple possible solutions: a. do not allow extra-repo uses. b. allow extra-repo uses, but force to pin shas. c. i've already started working on a permission system, where you explicitly specify which operations each module can do, not very unlike the AWS permissions model. Aside from these stuff, there is the issue where we need to know who even can modify the sky code - should it be all org members? select members? That's something I'd like to button down first.
3. Sure, though if the issue is money I'd be happy to just provide it for free for having you folk as gunieapigs :)

**Comment by cmluciano** [comment]
@itayd How difficult would it be for me to try to create a module? Can I just follow your pattern to create another bot?

**Comment by itayd** [comment]
@cmluciano you can try anything in this repo - but there are no docs yet. what do you want to try? there are a bunch of github functions supplied by repokitteh you can use. want to follow up on slack?

**Pull request closed without merging**